src/test/java/cmput402/tdd/AppTest.java

```diff
@@ -227,4 +228,257 @@ public void testDisplayPeople() {
         
     }
 
+    @Test
+    public void testEditRecipe(){
+        App app = new App();
+        Recipe recipe = new Recipe("apple pie");
+
+        Item item1 = new Item("apple",4);
+        Item item2 = new Item("sugar",2);
+
+
+        //Test 1: Add new item to recipe
+        ByteArrayInputStream in = new ByteArrayInputStream("add 1\napple\n4.0".getBytes());
+        Scanner scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Could not add item to recipe", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(1, recipe.items.get(item1).intValue());
+
+        //Test 2 Add an existing item to recipe again
+        in = new ByteArrayInputStream("add 3\napple\n4.0".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Could not add item to recipe", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(4, recipe.items.get(item1).intValue());
+
+        //Test 3 Add 0 quantity to recipe
+        in = new ByteArrayInputStream("add 0\nsugar\n2.0".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Could not add item to recipe", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(4, recipe.items.get(item1).intValue());
+        assertFalse(recipe.items.containsKey(item2));
+
+
+        //Test 4 Add another new item to recipe
```

Comment:
Again, if this is successful then we should fail in the catch, not assert something